src/Tribe/Aggregator/Processes/Import_Events.php

```diff
@@ -0,0 +1,365 @@
+<?php
+
+/**
+ * Class Tribe__Events__Aggregator__Processes__Import_Events
+ *
+ * Imports events in an async queue.
+ *
+ * @since TBD
+ */
+class Tribe__Events__Aggregator__Processes__Import_Events extends Tribe__Process__Queue {
+	/**
+	 * @var string
+	 */
+	protected $transitional_id;
+
+	/**
+	 * @var int The post ID of the record associated to this queue instance.
+	 */
+	protected $record_id;
+
+	/**
+	 * @var bool Whether the current item has dependencies or not.
+	 */
+	protected $has_dependencies = true;
+
+	/**
+	 * @var Tribe__Events__Aggregator__Record__Activity[]
+	 */
+	protected $activities = array();
+
+	/**
+	 * Returns the async process action name.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public static function action() {
+		return 'ea_import_events';
+	}
+
+	/**
+	 * Adds transitional data, used to check dependencies, to an event linked posts.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $event
+	 */
+	public function add_transitional_data( array $event ) {
+		$venue_id      = Tribe__Utils__Array::get( $event, 'EventVenueID', false );
+		$organizer_ids = Tribe__Utils__Array::get( $event, array( 'Organizer', 'OrganizerID' ), false );
+
+		if ( false !== $venue_id ) {
+			update_post_meta( $venue_id, $this->get_transitional_meta_key(), get_post_meta( $venue_id, '_tribe_aggregator_global_id', true ) );
+		}
+
+		if ( false !== $organizer_ids && is_array( $organizer_ids ) ) {
+			foreach ( $organizer_ids as $organizer_id ) {
+				update_post_meta( $organizer_id, $this->get_transitional_meta_key(), get_post_meta( $organizer_id, '_tribe_aggregator_global_id', true ) );
+			}
+		}
+	}
+
+	/**
+	 * Returns the `meta_key` that will be used to store the transitional data
+	 * in linked post for this import process.
+	 *
+	 * @since TBD
+	 *
+	 * @param null $transitional_id
+	 *
+	 * @return string
+	 */
+	public function get_transitional_meta_key( $transitional_id = null ) {
+		if ( null === $transitional_id ) {
+			$transitional_id = $this->transitional_id;
+		}
+
+		return '_tribe_import_' . $transitional_id;
+	}
+
+	/**
+	 * Sets the final part `meta_key` that should be used to store transitional
+	 * information for this import process.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $transitional_id
+	 */
+	public function set_transitional_id( $transitional_id ) {
+		$this->transitional_id = $transitional_id;
+	}
+
+	/**
+	 * Overrides the parent `save` method to save some additonal data.
+	 *
+	 * @since TBD
+	 *
+	 * @return Tribe__Events__Aggregator__Processes__Import_Events
+	 */
+	public function save() {
+		add_filter( "tribe_process_queue_{$this->identifier}_save_data", array( $this, 'save_data' ) );
```

Comment:
With this code sitting inside of the EA client, is that the [correct prefix](http://moderntribe.github.io/products-engineering/guidelines/plugins-namespacing/) (should it be `tribe_aggregator_`)?

Reply:
...ditto other hooks in this class

Reply:
This is an add_filter, the hook itself is running in tribe-common